pandas/core/indexing.py

```diff
@@ -602,7 +603,7 @@ def _get_setitem_indexer(self, key):
         """
         Convert a potentially-label-based key into a positional indexer.
         """
-        if self.name == "loc":
+        if self.name == "loc" or is_bool_dtype(np.array(key)):
```

Comment:
is it only lists we want here, or would `com.is_bool_indexer(key)` work?

Reply:
That is better, changed it :)